fasttld/FastTLDExtract.py

```diff
@@ -24,23 +48,24 @@ def update(self, *args, **kwargs):
     def nested_dict(self, dic, keys):
         """
         The idea of this function is based on https://stackoverflow.com/questions/13687924
-        :param dic: 
-        :param keys: 
-        :return: 
+        :param dic:
+        :param keys:
+        :return:
         """
+        end = False
         for key in keys[:-1]:
             dic_bk = dic
-            dic = dic.setdefault(key, {})
+            if key not in dic:
+                dic[key] = {}
+            dic = dic[key]
             if isinstance(dic, bool):
+                end = True
```

Comment:
Without this `end=True` flag, the test case in `test_nested_dict()` will have the dictionary include an erroneous extra `com: True`